src/Common/src/System/Net/CookieParser.cs

```diff
@@ -0,0 +1,892 @@
+                            switch (_tokenizer.Token)
+                            {
+                                case CookieToken.Comment:
+                                    if (!commentSet)
+                                    {
+                                        commentSet = true;
+                                        cookie.Comment = _tokenizer.Value;
+                                    }
+                                    break;
+
+                                case CookieToken.CommentUrl:
+                                    if (!commentUriSet)
+                                    {
+                                        commentUriSet = true;
+                                        Uri parsed;
+                                        if (Uri.TryCreate(CheckQuoted(_tokenizer.Value), UriKind.Absolute, out parsed))
+                                        {
+                                            cookie.CommentUri = parsed;
+                                        }
+                                    }
+                                    break;
+
+                                case CookieToken.Domain:
+                                    if (!domainSet)
+                                    {
+                                        domainSet = true;
+                                        cookie.Domain = CheckQuoted(_tokenizer.Value);
+                                        IsQuotedDomainField.SetValue(cookie, _tokenizer.Quoted);
+                                    }
+                                    break;
+
+                                case CookieToken.Expires:
+                                    if (!expiresSet)
+                                    {
+                                        expiresSet = true;
+
+                                        DateTime expires;
+                                        if (DateTime.TryParse(CheckQuoted(_tokenizer.Value),
+                                            CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, out expires))
+                                        {
+                                            cookie.Expires = expires;
+                                        }
+                                        else
+                                        {
+                                            // This cookie will be rejected
+                                            InternalSetNameMethod.Invoke(cookie, EmptyStringArray);
+                                        }
+                                    }
+                                    break;
+
+                                case CookieToken.MaxAge:
+                                    if (!expiresSet)
+                                    {
+                                        expiresSet = true;
+                                        int parsed;
+                                        if (int.TryParse(CheckQuoted(_tokenizer.Value), out parsed))
```

Comment:
Is that a breaking change? That means we'd now accept values of `CookieToken.MaxAge` that are doubles instead of integers only. Not sure that's intended